File: core/csharp/Exceptions/FilterValidationException.cs
using System;

namespace DynamicFilter.Core.Exceptions
{
    /// <summary>
    /// Exception thrown when a filter condition cannot be validated or constructed.
    /// </summary>
    public class FilterValidationException : Exception
    {
        /// <summary>
        /// Initializes a new instance of the FilterValidationException class.
        /// </summary>
        /// <param name="message">The error message</param>
        public FilterValidationException(string message) : base(message)
        {
        }

        /// <summary>
        /// Initializes a new instance of the FilterValidationException class.
        /// </summary>
        /// <param name="message">The error message</param>
        /// <param name="innerException">The inner exception</param>
        public FilterValidationException(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}

File: core/java/src/main/java/io/github/cyfko/filterql/core/exception/FilterValidationException.java
package io.github.cyfko.filterql.core.exception;

/**
 * Exception thrown when a filter condition cannot be validated or constructed.
 * <p>
 * This exception is used to explicitly indicate errors related to filter validation,
 * making it easier to handle specific errors in the filtering pipeline.
 * </p>
 *
 * <p>Usage example:</p>
 * <pre>{@code
 * if (!filter.isValid()) {
 *     throw new FilterValidationException("The filter contains invalid values");
 * }
 * }</pre>
 *
 * @author Frank KOSSI
 * @since 2.0.0
 */
public class FilterValidationException extends RuntimeException {

    /**
     * Creates an exception with an explanatory message.
     *
     * @param message the description of the cause of the exception
     */
    public FilterValidationException(String message) {
        super(message);
    }

    /**
     * Creates an exception with an explanatory message and an underlying cause.
     *
     * @param message the description of the cause of the exception
     * @param cause   the original cause of the exception
     */
    public FilterValidationException(String message, Throwable cause) {
        super(message, cause);
    }
}

